### ISSUE TITLE
starting live stream crashes Kodi

### ISSUE THREAD

**Issue by lebeno** [add-ons/plugin.video.vtm.go#353]
**Describe the bug**
When starting livestream any of the VTM channels, Kodi  immediately crashes and reboots before any video is being played. Tried multiples times.

**Logs**
```
2023-02-05 20:42:05.602 T:1490     INFO <general>: [plugin.video.vtm.go] [urllib3.connectionpool] https://dcs-live.mp.lura.live:443 "GET /server/play/nzq4mI2owEdun8GGa/manifest.mpd?encp=7g1aGC1V-u1cqV8y1zqI6Q:adXFAeUT2otitOI5cWxISU19tGr9XYdPFiBbc2UOE9PUehcX2h2NTUzIDmvsPjRq9olOJ2hWq2gEHor1Cy-3ch3CA5vxZI7RPt3MUfS6dNESBtfxOMk8YDpg-fSegc-OVdttLw4wnimAkBLrZiV5yrB9mPjaYUlXc-Sq5-SHhBbo2WmJrh3yUqtXr912nti2EZUmyUudTHwlVPa--XmRE3ll_Xd2w-xF-lqwqvi-0ZYhSSrbQAAa4tYtzhIR2bSTyDAdftwnVk9XNJuxT0dKIfQ3yK4WTnMRYFfsADGGvjI77LoGrEVkm2rQ8JDRw6m0QUeRIG8MHJCqOIm3CQdeP6qR6_8w3cdAVOJmyheGuVqn7Ml6B9IBQ47arzzV1xO0glhiMligfccQ_ZIyq69xk2p4B8GPb_pWnfzcu_qDNjjhxR6B7kY2NjGGScfhtOPGhDRDycTkDWF6xmwSyxAKynYcS3pxB2LWlXTj5Ksz-G_l9awWJNwBZ2iVjWX9V5eDtpJiS79bb6VzH3eNIMo0FHk8w26SZWUQX4Dtz_CeJogSxrgVaQ-wrFWeCT1E47xvij78Yxxba8wWGSmLU2PhxcIvYVCD5zAw14mSIQ6-ZB6udYcL4T4unLt7TdHs9QEWFMfnbc4CUFzxAWyCjhK71zPc7rPqwrF2tB12HcNTMjx1jTBHfCJ0Mr4Gi9vtT7p-gJIz1hXl_b9whc70TEIwLPJjaEaRh-pKqi-fy_eG6-kxEq63AXtb2pDM5zJ4a3BI_q1OaxWEPFMofQVcR8_6R1eb91Y2nE4MfnTLWDTTd_q9LOudojohCAes2l1Ld03AMCjUnMpqv63eLWzmE6z4sILzolcdugvi9vvDrINDXnTQbTxX6yn4PhXvBcQnF0GuFCuhYQteWXB9MMvD5RQ_ZxKEtQ-Rz-Tspj-TPtycWSEvnhdALK9IobV2MUmTKlBbTdMmo6gA-SbKc14o_2VE2jsCoj0zhgmMcq91GO3zgOOyYnUmT5faIMnSjTbcwLf-V9c6B9MWD7pUgHaq21pQ8g&anvtrid=kuo12dKLrF2Tiz9cOmRn68vsk7GiCdVA&anvauth=tb=0~te=1675626215~sgn=acf33c1906e6860353198091d95c96beccfbf6e28bf74570bd8922e9c10f85b8&t=1675626125 HTTP/1.1" 200 None
2023-02-05 20:42:05.606 T:1490     INFO <general>: Skipped 6 duplicate messages..
2023-02-05 20:42:05.606 T:1490     INFO <general>: [plugin.video.vtm.go] [resources.lib.vtmgo.util] Got response (status=200): <?xml version="1.0" encoding="utf-8"?>
                                                   <MPD xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                                                       xmlns="urn:mpeg:dash:schema:mpd:2011"
                                                       xmlns:xlink="http://www.w3.org/1999/xlink"
                                                       xsi:schemaLocation="urn:mpeg:DASH:schema:MPD:2011 http://standards.iso.org/ittf/PubliclyAvailableStandards/MPEG-DASH_schema_files/DASH-MPD.xsd"
                                                       profiles="urn:mpeg:dash:profile:isoff-main:2011"
                                                       type="dynamic"
                                                       availabilityStartTime="1970-01-01T00:00:00Z"
                                                       publishTime="2023-02-05T19:42:05Z"
                                                       minimumUpdatePeriod="PT05S"
                                                       suggestedPresentationDelay="PT10S"
                                                       minBufferTime="PT02S">
                                                       <Location>https://dcs-live-ue4.mp.lura.live/server/play/nzq4mI2owEdun8GGa/manifest.mpd?anvsid=m177635096-nee0580f0193bc9e3a6aeeb11ced9324b</Location>
                                                   </MPD>
2023-02-05 20:42:05.609 T:1490     INFO <general>: Skipped 6 duplicate messages..
2023-02-05 20:42:05.608 T:1490     INFO <general>: [plugin.video.vtm.go] [resources.lib.vtmgo.vtmgostream] Followed redirection from https://dcs-live.mp.lura.live/server/play/nzq4mI2owEdun8GGa/manifest.mpd?encp=7g1aGC1V-u1cqV8y1zqI6Q:adXFAeUT2otitOI5cWxISU19tGr9XYdPFiBbc2UOE9PUehcX2h2NTUzIDmvsPjRq9olOJ2hWq2gEHor1Cy-3ch3CA5vxZI7RPt3MUfS6dNESBtfxOMk8YDpg-fSegc-OVdttLw4wnimAkBLrZiV5yrB9mPjaYUlXc-Sq5-SHhBbo2WmJrh3yUqtXr912nti2EZUmyUudTHwlVPa--XmRE3ll_Xd2w-xF-lqwqvi-0ZYhSSrbQAAa4tYtzhIR2bSTyDAdftwnVk9XNJuxT0dKIfQ3yK4WTnMRYFfsADGGvjI77LoGrEVkm2rQ8JDRw6m0QUeRIG8MHJCqOIm3CQdeP6qR6_8w3cdAVOJmyheGuVqn7Ml6B9IBQ47arzzV1xO0glhiMligfccQ_ZIyq69xk2p4B8GPb_pWnfzcu_qDNjjhxR6B7kY2NjGGScfhtOPGhDRDycTkDWF6xmwSyxAKynYcS3pxB2LWlXTj5Ksz-G_l9awWJNwBZ2iVjWX9V5eDtpJiS79bb6VzH3eNIMo0FHk8w26SZWUQX4Dtz_CeJogSxrgVaQ-wrFWeCT1E47xvij78Yxxba8wWGSmLU2PhxcIvYVCD5zAw14mSIQ6-ZB6udYcL4T4unLt7TdHs9QEWFMfnbc4CUFzxAWyCjhK71zPc7rPqwrF2tB12HcNTMjx1jTBHfCJ0Mr4Gi9vtT7p-gJIz1hXl_b9whc70TEIwLPJjaEaRh-pKqi-fy_eG6-kxEq63AXtb2pDM5zJ4a3BI_q1OaxWEPFMofQVcR8_6R1eb91Y2nE4MfnTLWDTTd_q9LOudojohCAes2l1Ld03AMCjUnMpqv63eLWzmE6z4sILzolcdugvi9vvDrINDXnTQbTxX6yn4PhXvBcQnF0GuFCuhYQteWXB9MMvD5RQ_ZxKEtQ-Rz-Tspj-TPtycWSEvnhdALK9IobV2MUmTKlBbTdMmo6gA-SbKc14o_2VE2jsCoj0zhgmMcq91GO3zgOOyYnUmT5faIMnSjTbcwLf-V9c6B9MWD7pUgHaq21pQ8g&anvtrid=kuo12dKLrF2Tiz9cOmRn68vsk7GiCdVA&anvauth=tb=0~te=1675626215~sgn=acf33c1906e6860353198091d95c96beccfbf6e28bf74570bd8922e9c10f85b8&t=1675626125 to https://dcs-live-ue4.mp.lura.live/server/play/nzq4mI2owEdun8GGa/manifest.mpd?anvsid=m177635096-nee0580f0193bc9e3a6aeeb11ced9324b
2023-02-05 20:42:05.647 T:1490     INFO <general>: Skipped 6 duplicate messages..
2023-02-05 20:42:05.647 T:1490     INFO <general>: [plugin.video.vtm.go] [resources.lib.kodiplayer] Player: Waiting for playback
2023-02-05 20:42:05.648 T:1490     INFO <general>: Skipped 6 duplicate messages..
2023-02-05 20:42:05.648 T:1490     INFO <general>: [plugin.video.vtm.go] [resources.lib.kodiplayer] Player: Not playing
2023-02-05 20:42:05.715 T:1315     INFO <general>: Skipped 13 duplicate messages..
2023-02-05 20:42:05.715 T:1315     INFO <general>: VideoPlayer::OpenFile: pvr://channels/tv/Alle%20kanalen/pvr.iptvsimple_762788450.pvr
2023-02-05 20:42:05.718 T:1564     INFO <general>: Creating InputStream
2023-02-05 20:42:05.719 T:1564     INFO <general>: AddOnLog: inputstream.adaptive: SetVideoResolution (1920 x 1080)
2023-02-05 20:42:05.724 T:1564    ERROR <general>: GetDirectory - Error getting /usr/lib/kodi/addons/inputstream.adaptive/
```

**System**
 - Addon version: 1.4.0+matrix.1
 - Kodi version: Kodi (Matrix) v19.5
 - Inputstream adaptive version: 19.0.7.1
 - Operating System (Windows / Mac OS / Android / LibreElec / OSMC / ...): LibreELEC 10.0.4
 - Special Hardware (RPI / Vero4K+ / ...): RasPi 4

**Additional context**
 - Tried clearing the cache in the VTM GO plugin settings, but that does not resolve the issue.
 - Streaming VRT MAX works without issues.

**Comment by pborreke** [comment]
I've the same issue as described above, it's happening since today, 5/02/2023.

2023-02-05 21:45:05.753 T:11028    INFO <general>: CAddonMgr::ADDON::CAddonMgr::FindAddons: plugin.video.vtm.go v1.4.0+matrix.1 installed

2023-02-05 21:45:17.002 T:11228    INFO <general>: CPythonInvoker(5, C:\Users\pborr\AppData\Roaming\Kodi\addons\plugin.video.vtm.go_old\addon_entry.py): script successfully run
2023-02-05 21:45:18.560 T:11228    INFO <general>: initializing python engine.
2023-02-05 21:45:18.818 T:11228    INFO <general>: CPythonInvoker(5, C:\Users\pborr\AppData\Roaming\Kodi\addons\plugin.video.vtm.go_old\addon_entry.py): script successfully run
2023-02-05 21:45:20.024 T:11228    INFO <general>: initializing python engine.
2023-02-05 21:45:20.385 T:11228    INFO <general>: CPythonInvoker(5, C:\Users\pborr\AppData\Roaming\Kodi\addons\plugin.video.vtm.go_old\addon_entry.py): script successfully run
2023-02-05 21:45:21.734 T:11228    INFO <general>: initializing python engine.
2023-02-05 21:45:22.203 T:11228 WARNING <general>: [script.module.inputstreamhelper] Widevine update check was made on 2023-01-13 16:03
2023-02-05 21:45:22.513 T:14344    INFO <general>: CPythonInvoker(3, C:\Users\pborr\AppData\Roaming\Kodi\addons\service.xbmc.versioncheck\resources\lib\runner.py): script successfully run
2023-02-05 21:45:22.586 T:14344    INFO <general>: Python interpreter stopped
2023-02-05 21:45:22.836 T:11228   ERROR <general>: [plugin.video.vtm.go] [resources.lib.vtmgo.vtmgostream] No manifest url found at https://dcs-live.mp.lura.live/server/play/nzq4mI2owEdun8GGa/manifest.mpd?encp=sWKybcU1yAE6IJBLfA71Rw:2SKjEpGHuJLXBp5qZ_urvqxD83iChNP2WB_b3DE5L2oZ77hWbMN_tqw0R4Vc7M2tloa-mrfSS2OVnjd-7ulkTH5GtxKee1GHHc5cdac73EmbILu-b-5ZZrYEbklgmIM5MHaoElUjUOoSDqBiHrJke_Fkna0d8hrlIW_zRlh8dF197DSbBP6CGUb5OVKjfe2E24aLuOjDOfdWDDB_0tBlVpYWQriPKrgUb0Af6UkqgiUtcay0h7av6WuJYmQT1VobPKFW_VaLj2lOZpRzhX8zI99LRAdehB3bgz1guv9brby1HolboKTToWapxtYRxpA4axLR6XzOxwo-6mmQw8DdN1nSF_nx0xgIuFE8yGfzeuQnmY382xHkyrUKS_tMii_uc0b4Kx5rSQ5qU6zw7YVoklpR01vW8lTccOyQ_sskOyvqrvVvBsdmVHVZg41WYvvsAKNeftjM_F1yh_cuJJEnln11Wic8GVoMwoEPoNurel4kvW_AMF3xBHuJvkOE400_crDQQhvonVfNKMCWWu6U69uKveZ2Q8AedKnASP8rjjH0fwXrtz-vfzZynDvn9KrZH8L5qVSHJpIbo6yc-yoQjR0JC4jj5AQxVaMdRqy5gDkJjPMH5uH2HTJ-bupTgVCll2o9nliYqL7HxdekmRgJVeQJiJIcN-OhR505d3RwNes160bFjs1_55mrgmhLc9_Nu5gqHF6nxvcJ5olx6nTfF09geNM-fk8hvthJhtU0ikM8ERNe1NwXvw-jmBDI6nFMtiihDWmLCFU3Xmxm8qISgbCODTZlT0_56gcM57D-560Ct4znx3hXZgITJvi_ioSQimyU429-NDVZ0QuuWUc8Un6A6TBb4Z_xt6lXJpWN-ZQc38WAsBGUjneAUqZ94-Ba7iuktD4F0vlg743gcJOfvgW4QENXVHxFKPtaCm14OZWP0BLHhTPOXQNlg1uxM1EZSIgwPQgQsOpeKKR-9OxWSC19qBAjDFab5wXh1QZs_9ekf14CTWR4I21lPg905YNtJCnUMhFwgZn85BkRR88ZNA&anvtrid=iD903j0M2e48x0aLHP9yNai7UJBbMxPd&anvauth=tb=0~te=1675630013~sgn=c5f7bd722ca0dff969d195effb8a47431dfb1069feef5eaa4690228c6471e174&t=1675629923
2023-02-05 21:45:22.994 T:11028    INFO <general>: Skipped 3 duplicate messages..
2023-02-05 21:45:22.994 T:11028    INFO <general>: VideoPlayer::OpenFile: plugin://plugin.video.vtm.go/play/catalog/channels/d8659669-b964-414c-aa9c-e31d8d15696b?.pvr
2023-02-05 21:45:22.997 T:7028     INFO <general>: Creating InputStream
2023-02-05 21:45:23.010 T:7028     INFO <general>: AddOnLog: inputstream.adaptive: SetVideoResolution (1280 x 1024)
2023-02-05 21:45:23.012 T:7028    ERROR <general>: XFILE::CDirectory::GetDirectory - Error getting C:\Program Files\Kodi\addons\inputstream.adaptive\

**Comment by michaelarnauts** [comment]
Hmm, odd. Didn't see that yesterday when I made the release.

Also, a crashing Kodi is never the issue of an Add-on, since Kodi should basically always accept what an add-on throws at it. This seems like a bug in Kodi.

Have you tried upgrading to Kodi 20?

**Comment by cometoluc** [comment]
Hello, i have the same problem sins today but i am still on kodi 19.3(19.3.0)

I am scared to go to kodi 20
If i install kodi 20 over kodi19... must i than reinstall all my addons? And will my settings in kodi stay the same?

**Comment by cometoluc** [comment]
@michaelarnauts i will wait until there is maybe a new release of vtmgo on off the following days.
i tried the addon from vrtmax and there the live channels work like thy should. For comparrising reasons.

ik werk op windows 732bit laatste
vtmgo versie 1.4+matrix.1
kodi versie 19.3(19.3.0)

**Comment by mediaminister** [comment]
It seems VTM GO Popcorn SDK 6 serves a new livestream format that is not supported in Kodi 19.
Adding a fallback to Popcorn SDK 5 for Kodi 19 fixes this.
Please test:
Kodi 19/20: [plugin.video.vtm.go-1.4.0+matrix.1.zip](https://github.com/add-ons/plugin.video.vtm.go/files/10659964/plugin.video.vtm.go-1.4.0%2Bmatrix.1.zip)
Kodi 18: [plugin.video.vtm.go-1.4.0.zip](https://github.com/add-ons/plugin.video.vtm.go/files/10659963/plugin.video.vtm.go-1.4.0.zip)

**Comment by cometoluc** [comment]
do you want to say that i have to reinstall with zip file and that the updated version was no good?

**Comment by mediaminister** [comment]
Yes, unfortunately, livestreaming in the updated version was only tested on Kodi 20.
So, it's very important that as many as possible people test the new zip file above before another official release can be made.

**Comment by cometoluc** [comment]
Hello, if i install kodi 20 will i 
keep my data, do i have to install all my addons again or can i best wait for a few days when it's fixed in kodi 19....

**Comment by cometoluc** [comment]
i installed the zipfile and now the live works again in kodi versie 19.3(19.3.0)  Thanks

ik werk op windows 732bit laatste
vtmgo versie 1.4+matrix.1
kodi versie 19.3(19.3.0)

**Comment by lebeno** [comment]
> It seems VTM GO Popcorn SDK 6 serves a new MPEG-DASH livestream manifest that is not supported in Kodi 19 and/or Inputstream Adaptive for Kodi 19. Adding a fallback to Popcorn SDK 5 for Kodi 19 fixes this. Please test: Kodi 19/20: [plugin.video.vtm.go-1.4.0+matrix.1.zip](https://github.com/add-ons/plugin.video.vtm.go/files/10659964/plugin.video.vtm.go-1.4.0%2Bmatrix.1.zip) Kodi 18: [plugin.video.vtm.go-1.4.0.zip](https://github.com/add-ons/plugin.video.vtm.go/files/10659963/plugin.video.vtm.go-1.4.0.zip)

Tested on my install (Kodi 19.5, LibreELEC 10.0.4, RasPi 4) and it works. Thank you!

**Comment by peno64** [comment]
Also had the crash of kodi with livestream. Installed the zip for matrix on kodi 19.5 on a rpi4 with libreelec and it works.
Thx

**Comment by peno64** [comment]
> Hello, if i install kodi 20 will i
> keep my data, do i have to install all my addons again or can i best wait for a few days when it's fixed in kodi 19....

Yes you will keep all your data and your addons will still work. However many people complain about crashes with kodi 20.  Especially if you use builds/special skins/many addons. This seems not be a real kodi issue but a problem in the python version linked with kodi. A next version should solve that problem.

Some repositories are also not compatible with kodi 20 because kodi 20 is less forgiving about old/obsolete syntax used in repo definitions.

So it is a bit of a risk to update to kodi 20

**Comment by nicspep** [comment]
Hello, 

I have Kodi20, nexus and cannot fix the issue with VTM Go. 
I tried to unzip the above file, but it's not working? 
Is there something I oversee or not know? 
I'm a bit new in these things..

Your help would be appreciated. 

Thanks.

**Comment by Dominiquedb** [comment]
> Hello,
> 
> I have Kodi20, nexus and cannot fix the issue with VTM Go. I tried to unzip the above file, but it's not working? Is there something I oversee or not know? I'm a bit new in these things..
> 
> Your help would be appreciated.
> 
> Thanks.

Don't unzip it. Go to addons, install from zip file, and choose the zip file to install it in kodi.

**Comment by nicspep** [comment]
Hi Dominique, 

This is what's strange? in the past I've done this before with issues for Play4.
But now in addons even unzipped, Kodi doesn't recognize the zip file? 
When I go install from zip, I don't see the zip file?

**Comment by peno64** [comment]
You are probably on android. In that case you must give all permissions to the kodi app in android. kodi 20 uses the latest android toolkit and it is more restrictive on permissions and you must explicitely allow the app to access files.

**Comment by nicspep** [comment]
Good evening peno64.
Indeed it was a Android issue. this is solved, thanks for that.
However I unzipped and the addon updateed but still the error VTMGO Error.

**Comment by peno64** [comment]
@nicspep unzip is not the way to update. In kodi you must choose install from zip and choose the ziip file and restart kodi to be sure.

**Comment by nicspep** [comment]
@ peno64 sorry, I did explain myself wrongly... I mean I managed to use the zip file and the addon updated.
However without result. the issue remains? 
Could it be that the problem is Kodi 2à Nexus? I read here in the comments that the most of the users is using kodi 19?
were the problem is solved after the update.

**Comment by nicspep** [comment]
@ peno64 started from scratch. It worked! Thanks for the support !

**Comment by pborreke** [comment]
I've installed the zip on my kodi 19.5 and it works as before the issue, thanks for the solution!

**Comment by michaelarnauts** [comment]
Fixed in #354. I'll make a new release now.

Thanks again to @mediaminister for the fix, and you all for testing it! ❤️

I don't have as much free time as I used to have when I created this addon, so I'm sorry for the delay in responding and releasing fixed versions.
